src/data/value/literal.rs

```diff
@@ -265,6 +281,13 @@ fn parse_time(v: &str) -> Result<NaiveTime> {
     Err(ValueError::FailedToParseTime(v).into())
 }
 
+fn parse_uuid(v: &str) -> Result<u128> {
+    match Uuid::parse_str(v) {
+        Ok(u) => Ok(u.as_u128()),
+        _ => Err(ValueError::FailedToParseUUID(v.to_owned()).into()),
```

Comment:
This error type is always crushed.

```rust
        _ => Err(ValueError::FailedToParseUUID(v.to_owned()).into()),
```

Reply:
I don't understand this comment, can you explain more detail?

Reply:
Ummm... it seems deprecated
For example, if an error occurs, the function is being used and changed to a different default value

Reply:
ex) For example, chang from error type to default in this way
```rust
match parse_uuid(r) {
                Ok(r) => l == &r,
                Err(_) => false,
}
```

Q: Should it default to when parsing fails? Or do you just want to spit out an error?

Reply:
This function is not deprecated and still use in here.
[PartialEq](https://github.com/gluesql/gluesql/pull/360/files#diff-06ff4b2f6cc1883c8536124b289f4e2beb48ceae7a80ce995f036000fa7a2229R45
)
[PartialOrd](https://github.com/gluesql/gluesql/pull/360/files#diff-06ff4b2f6cc1883c8536124b289f4e2beb48ceae7a80ce995f036000fa7a2229R82)

I think this function spit out an error when parsing fails, and other clients using this function decide how to handle this error. ex) also spit out error, using default value...

So, like the above links, i used this function that handle errors in the clients.
Is this anti pattern in rust?

Reply:
Already in `Uuid::parse_str` there is an error type for parsing failure.
I don't think it's necessary to crush this type twice.
For example in Value
```rust
            (DataType::UUID, Literal::Text(v)) => parse_uuid(v)
                .map(Value::UUID)
                .map_err(|_| ValueError::FailedToParseUUID(v.to_string()).into()),
```
This code encounters a type called `FailedToParseUUID` twice. (in parse_uuid, now)
I don't quite understand this part

It would be nice to get help from @panarch
I will request a review.

Reply:
The reason I created this function is because our UUID data type is u128, so I wanted to collect the string parsing and converting logic in one place rather than in several places. (But i missed several places using Uuid::parse_str in tests)

But I agree your comments It is unnecessary wrapping function and crush error twice.
If @panarch agree this function unnecessary, I will change this.